sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DocumentCollection.java

```diff
@@ -90,6 +90,7 @@ public void setIndexingPolicy(IndexingPolicy indexingPolicy) {
         }
 
         this.indexingPolicy = indexingPolicy;
+        setProperty(this, Constants.Properties.INDEXING_POLICY, this.indexingPolicy);
```

Comment:
This was necessary because the container replace method that is being used in azure-cosmos depends on the __self link being in the propertyBag. You can see below in SimpleCosmosRepository that I have to retrieve the CosmosContainerProperties from the server so that this field is contained in the properties bag. I cannot use the one that i create myself using CosmosEntityInformation since it will not contain the __self link. 
So, I retrieve CosmosContainerProperties from the server and then set the new indexing policy on it. This change makes it so that calling setIndexingPolicy updates that underlying json representation that is sent to the server. Without this is place, calling setIndexingPolicy() has no effect.

Reply:
@kushagraThapar  @Blackbaud-EricSlater 
There is a test in `com.azure.cosmos.CosmosContainerTest.replace` validating replace scenario. Does this mean that test is broken?

Reply:
im not sure what i missed in my testing that made me think this was necessary, but it is working without it. The DocumentCollection is re-serialized when getV2Collection() is called on the DocumentCollection in CosmosAsyncContainer.replaceInternal. I removed the edits to this file.

However, on a separate note, in looking at the test you mentioned, i did notice that it was not actually changing the index policy to a new value in the test. It was creating a container with indexMode.CONSISTENT and replacing it with a container with indexMode.CONSISTENT, which is also the default value if no policy is specified. This is not a very good validation that the indexPolicy is actually being replaced.
I updated the test to replace the container with a different index policy (automatic = false and indexMode.NONE). This should be stronger validation

Reply:
thank you @Blackbaud-EricSlater 

@kushagraThapar please run the full CI.

Reply:
Thanks @Blackbaud-EricSlater for updating the tests, however on the spring side, can you please add the support for CosmosReactiveRepository as well ?
Otherwise, when using reactiveRepositoies, the indexing policy update will not take affect.

Reply:
will do